script/nan-spec-runner.js

```diff
@@ -31,25 +31,44 @@ async function main () {
   const cxx = path.resolve(clangDir, 'clang++');
   const ld = path.resolve(clangDir, 'lld');
 
+  const platformSpecificFlags = [];
+  if (process.platform === 'darwin') {
+    const sdkPath = path.resolve(BASE, 'out', utils.getOutDir(), 'sdk', 'xcode_links');
+    const possibleSdks = (await fs.promises.readdir(sdkPath)).filter(fileName => fileName.endsWith('.sdk'));
+    const sdkToUse = possibleSdks[0];
+    if (!sdkToUse) {
+      console.error('Could not find an SDK to use for the NAN tests');
+      process.exit(1);
+    }
+    if (possibleSdks.length) {
+      console.warn(`Multiple SDKs found in the xcode_links directory, using the first one we found ${sdkToUse}`);
+    }
+    platformSpecificFlags.push(
+      `-isysroot ${path.resolve(sdkPath, sdkToUse)}`
+    );
+  }
+
   // TODO(ckerr) this is cribbed from read obj/electron/electron_app.ninja.
   // Maybe it would be better to have this script literally open up that
   // file and pull cflags_cc from it instead of using bespoke code here?
   // I think it's unlikely to work; but if it does, it would be more futureproof
   const cxxflags = [
     '-std=c++17',
     '-nostdinc++',
-    `-isystem"${path.resolve(BASE, 'buildtools', 'third_party', 'libc++')}"`,
+    `-I"${path.resolve(BASE, 'buildtools', 'third_party', 'libc++')}"`,
```

Comment:
a bit surprising that this changes from `-isystem` to `-I`, what's the reasoning there?

Reply:
`-isystem` is more correct in the abstract. If something broke in the roll to require this change I'd be interested in what it was